From bf05fd051b225488bafdd30076f5e482c19b69d8 Mon Sep 17 00:00:00 2001
From: Jean-Christophe Trotin <jean-christophe.trotin@st.com>
Date: Fri, 21 Oct 2016 14:35:50 +0200
Subject: [PATCH 21/23] v4l2transform: keep colorimetry and chroma-site fields
 if passthrough

Same patch as the ones applied to videoconvert and glcolorconvert
(see https://bugzilla.gnome.org/show_bug.cgi?id=748141).

Change-Id: I2efd4e1557f6461d34d2f0ca63c7f5df70e6bee3
Reviewed-on: https://gerrit.st.com/59458
Reviewed-by: Jean Christophe TROTIN <jean-christophe.trotin@st.com>
Tested-by: Jean Christophe TROTIN <jean-christophe.trotin@st.com>
---
 sys/v4l2/gstv4l2transform.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/sys/v4l2/gstv4l2transform.c b/sys/v4l2/gstv4l2transform.c
index 52a29f4..3b17c1b 100644
--- a/sys/v4l2/gstv4l2transform.c
+++ b/sys/v4l2/gstv4l2transform.c
@@ -866,6 +866,12 @@ done:
   /* fixate remaining fields */
   othercaps = gst_caps_fixate (othercaps);
 
+  if (direction == GST_PAD_SINK) {
+    if (gst_caps_is_subset (caps, othercaps)) {
+      gst_caps_replace (&othercaps, caps);
+    }
+  }
+
   return othercaps;
 }
 
-- 
2.7.4

